src/messaging/ExchangeContext.cpp

```diff
@@ -194,7 +264,6 @@ void ExchangeContext::Reset()
 ExchangeContext * ExchangeContext::Alloc(ExchangeManager * em, uint16_t ExchangeId, uint64_t PeerNodeId, bool Initiator,
                                          ExchangeDelegate * delegate)
 {
-    VerifyOrDie(delegate != nullptr);
```

Comment:
We need to create a temporary exchange to send ack for a duplicate message since the previous exchange might be closed already after the node receive the first message. In this case, the temporary exchange won't have a delegate

Reply:
Right, that's what I found.  The point is that sort of thing should go in commit messages...